app/src/test/java/org/mozilla/fenix/historymetadata/interactor/HistoryMetadataInteractorTest.kt

```diff
@@ -0,0 +1,112 @@
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.R
+import org.mozilla.fenix.historymetadata.HistoryMetadataGroup
+import org.mozilla.fenix.historymetadata.controller.HistoryMetadataController
+import org.mozilla.fenix.home.recentbookmarks.controller.RecentBookmarksController
+import org.mozilla.fenix.home.recenttabs.controller.RecentTabController
+import org.mozilla.fenix.home.sessioncontrol.DefaultSessionControlController
+import org.mozilla.fenix.home.sessioncontrol.SessionControlInteractor
+import org.mozilla.fenix.home.sessioncontrol.viewholders.pocket.PocketStoriesController
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class HistoryMetadataInteractorTest {
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule(testDispatcher)
+
+    private val navController = mockk<NavController>(relaxed = true)
+    private val defaultSessionControlController: DefaultSessionControlController =
+        mockk(relaxed = true)
+    private val recentTabController: RecentTabController = mockk(relaxed = true)
+    private val recentBookmarksController: RecentBookmarksController = mockk(relaxed = true)
+    private val pocketStoriesController: PocketStoriesController = mockk(relaxed = true)
+    private val historyMetadataController: HistoryMetadataController = mockk(relaxed = true)
+
+    private lateinit var interactor: SessionControlInteractor
+
+    @Before
+    fun setup() {
+        every { navController.currentDestination } returns mockk {
+            every { id } returns R.id.homeFragment
+        }
+
+        interactor = SessionControlInteractor(
+            defaultSessionControlController,
+            recentTabController,
+            recentBookmarksController,
+            historyMetadataController,
+            pocketStoriesController
+        )
+    }
+
+    @After
+    fun cleanUp() {
+        testDispatcher.cleanupTestCoroutines()
+    }
+
+    @Test
+    fun onHistoryMetadatGroupClicked() {
+        val historyEntry =
+            HistoryMetadataGroup(
+                title = "mozilla",
+                historyMetadata = listOf(
+                    HistoryMetadata(
+                        key = HistoryMetadataKey("http://www.mozilla.com", null, null),
+                        title = "mozilla",
+                        createdAt = System.currentTimeMillis(),
+                        updatedAt = System.currentTimeMillis(),
+                        totalViewTime = 10,
+                        documentType = DocumentType.Regular,
+                        previewImageUrl = null
+                    )
+                )
+            )
+
+        interactor.onHistoryMetadataGroupClicked(historyEntry)
+        verify {
+            historyMetadataController.handleHistoryMetadataGroupClicked(historyEntry)
+        }
+    }
+
+    @Test
+    fun onHistoryMetadataShowAllClicked() {
```

Comment:
We actually have tests for this under SessionControlInteractorTest https://searchfox.org/mozilla-mobile/source/fenix/app/src/test/java/org/mozilla/fenix/home/SessionControlInteractorTest.kt#175

Reply:
that's on my todo for the tests :)